src/main.c

```diff
@@ -283,9 +304,13 @@ main(int argc, char **argv)
 
     for (int i = 0; i < r_c_threads; ++i)
         pthread_join(c_thread[i], &res);
+    consume_process_done();
 
     for (int i = 0; i < r_p_threads; ++i)
         pthread_join(p_thread[i], &res);
+    /* produce_process_done();
+     * not needed atm as the stat_thread checks for consumer state only.
+     */
```

Comment:
if not needed but doesn't hurt doing it, just reenable it.

Reply:
I'll enable this. It does not hurt really since this runs on exit.